x/auth/ante/ante.go

```diff
@@ -2,32 +2,45 @@ package ante
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
 	"github.com/cosmos/cosmos-sdk/x/auth/types"
 )
 
+// HandlerOptions are the options for ante handler build
+type HandlerOptions struct {
+	FeegrantKeeper  FeegrantKeeper
+	SigGasConsumer  func(meter sdk.GasMeter, sig signing.SignatureV2, params types.Params) error
+	SignModeHandler authsigning.SignModeHandler
```

Comment:
So because it's hard to have a sane default for this, let's move it back to a positional parameter.